README.md

```diff
@@ -1,2 +1,140 @@
+![Spectre](https://user-images.githubusercontent.com/1897842/31115297-0fe2c3aa-a822-11e7-90e6-92ceccf76137.jpg)
+
 # analysis-api
-Worker for manipulation of analyses.
+
+API for accessing details of analyses, their configurations and scheduling. Its
+main feature is high extensibility of the system through automated discovery
+and integration of new analysis modules.
+
+# Specification of a Worker
+
+In order to make worker discoverable and useful in the system, following
+specification must be met.
+
+## Input Specification Endpoint
+
+Endpoint `/schema/inputs/<task-name>` should provide simplistic specification
+of a computational task input as a [JSON-schema](http://json-schema.org/).
+
+Expected HTTP return code is `200`. If task of the given name does not exist,
+return code is `404`. Supported HTTP method is `GET`.
+
+For such minimal REST API, [Flask framework](flask.pocoo.org/) can be used.
+
+## Output Specification Endpoint
+
+Endpoint `/schema/outputs/<task-name>` should provide specification of a
+computational task output as follows:
+
+```
+[
+    {
+        "aspect": "some_aspect",
+        "friendly_name": "User-readable aspect name (will be displayed)",
+        "query_format":
+        {
+            // some JSON-schema
+        },
+        "output_type": "summary"
+    },
+    ...
+]
+```
+
+Expected HTTP return code is `200`. If task of the given name does not exist,
+return code is `404`. Supported HTTP method is `GET`.
+
+Explanation:
+
+* **aspect** defines aspect of the result we can query about. E.g. for DiviK
+it would be *summary* (with mean number of spectra in cluster, depth of analysis, etc.), *visualization* (a plot with segmentation marked). Value of
+`aspect` field must correspond to an endpoint defined as a part of REST API.
+More details below.
+* `friendly_name` - user-readable name, that will be displayed to user
+* `query_format` - defines, what parameters should be defined to obtain summary
+of the analysis. It should follow the format of JSON-schema
+* `output_type` - one of the following: `table` or `plot`
+
+Output types are described further [below](#output-types).
+
+## Result Access Endpoint
+
+Endpoint `/results/<task-name>/<aspect>` should handle `POST` requests with
```

Comment:
Why POST?

Reply:
Since in some cases we need parameterization to limit the amount of data sent to user. This applies e.g. to `level` of displayed segmentation in the case of DiviK.

Reply:
If endpoint return response with data it should be GET and parameters can be set in URL e.g /results/<task-name>/<aspect>?level=1
POST sends data to server to save.
But it is contract, I don't argue if you left POST :D